BuildMasterAutomation/Functions/Set-BMVariable.ps1

```diff
@@ -98,6 +110,11 @@ function Set-BMVariable
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
+    if ($Value -is [hashtable] -or $Value -is [array])

```

Comment:
Why do we need this check. The `$Value` should always be passed to `ConvertTo-BMOtterScriptExpression`, even if it's just a string because perhaps even the string may need some conversion (like escaping characters).